### PR TITLE
fix: destroy cleanup process

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Cleanup the listeners on `destroy` and use `eventManager` for event registration all over the project.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/fullscreen/fullscreen-controller.js [hunk 1]
```diff
@@ -255,6 +255,14 @@ class FullscreenController {
       this._player.dispatchEvent(new FakeEvent(this._player.Event.EXIT_FULLSCREEN));
     }
   }
+  /**
+   * Destroys the FullScreenController.
+   * @returns {void}
+   * @public
+   */
+  destroy(): void {
+    this._player = null;
```
Comment (yairans) [inline src/fullscreen/fullscreen-controller.js:264]: `this._isInBrowserFullscreen = false;`

Comment (yairans) [reply src/fullscreen/fullscreen-controller.js:264]: also need to clean the event manager here

Comment (Yuvalke) [reply src/fullscreen/fullscreen-controller.js]: I'm not sure it needs to be in this scope of change, I just detach the player refs all over the code.

File: src/engines/engine-decorator.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ class EngineDecorator extends FakeEventTarget implements IEngineDecorator {
   constructor(engine: IEngine, pluginWithDecorators: Array<IEngineDecoratorProvider>) {
     super();
     this._eventManager = new EventManager();
+    this._eventManager.listen(engine, EventType.PLAYER_DESTROY, () => this._destroy());
```
Comment (yairans) [inline src/engines/engine-decorator.js:25]: the engine doesn't trigger `PLAYER_DESTROY`

**Pull request merged**